app/src/main/java/app/tivi/inject/NetworkModule.kt

```diff
@@ -16,21 +16,64 @@
 
 package app.tivi.inject
 
+import android.content.Context
 import app.tivi.BuildConfig
+import coil.util.CoilUtils
 import dagger.Module
 import dagger.Provides
+import java.util.concurrent.TimeUnit
 import javax.inject.Singleton
+import okhttp3.ConnectionPool
+import okhttp3.Dispatcher
+import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
+import okhttp3.logging.LoggingEventListener
 
 @Module
 class NetworkModule {
     @Singleton
     @Provides
-    fun provideHttpLoggingInterceptor(): HttpLoggingInterceptor {
+    fun provideHttpLoggingInterceptor(): HttpLoggingInterceptor? {
+        if (!BuildConfig.DEBUG) {
+            return null
+        }
+
         return HttpLoggingInterceptor().apply {
-            if (BuildConfig.DEBUG) {
-                level = HttpLoggingInterceptor.Level.HEADERS
-            }
+            level = HttpLoggingInterceptor.Level.HEADERS
+        }
+    }
+
+    @Singleton
+    @Provides
+    fun provideHttpEventListener(): LoggingEventListener.Factory? {
+        if (!BuildConfig.DEBUG) {
+            return null
         }
+
+        return LoggingEventListener.Factory()
+    }
+
+    @Singleton
+    @Provides
+    fun provideOkHttpClient(
+        httpLoggingInterceptor: HttpLoggingInterceptor?,
+        loggingEventListener: LoggingEventListener.Factory?,
+        applicationContext: Context
+    ): OkHttpClient {
+        return OkHttpClient.Builder()
+            .apply {
+                if (httpLoggingInterceptor != null) {
+                    addInterceptor(httpLoggingInterceptor)
+                }
+                if (loggingEventListener != null) {
+                    eventListenerFactory(loggingEventListener)
+                }
+            }
+            .cache(CoilUtils.createDefaultCache(applicationContext))
```

Comment:
I think we should just create our own using `Cache(File(cacheDir, "http_cache"), 40 * 1024 * 1024))`. 

This needs to cater for all HTTP requests and the image cache, so relying on Coil's default is going to be too small.

Reply:
I think Coil is already using something like

```
        val size = (0.02 * diskSize).coerceIn(10.0, 250.0)
```

coil.util.Utils

```
    private const val MIN_DISK_CACHE_SIZE: Long = 10 * 1024 * 1024 // 10MB
    private const val MAX_DISK_CACHE_SIZE: Long = 250 * 1024 * 1024 // 250MB

    private const val DISK_CACHE_PERCENTAGE = 0.02
```

Want me to just bump the min to 40mb?

Reply:
Hmmm, I just don't want to depend on a Coil library implementation as a general caching mechanism.

Can we copy over the Coil implementation?

Reply:
Personally, I usually don't set a `Cache` on my base `OkHttpClient` and instead set separate cache files for my Retrofit OkHttp instance and my Coil OkHttp instance. For image-heavy apps network responses can be evicted from the cache pretty quickly if it's shared with Coil.

Reply:
@colinrtwhite OkHttp docs point users to using a single shared instance.  Your suggestion makes a lot of sense.  But does leave me wondering whether best option is

1) single instance/cache but appropriately sized for both
2) chained instance as you suggest, with only the cache differing.

In this case the images are appropriately sized off the network, and more likely to be cacheable than a feed type app with constant churn of images.  So I think both are valid.  Crucially with this change we still have the ability to do things like connection coalescing when the image host is actually the same as the api host.

If the second is clearly better it might imply updating the OkHttp docs "Customize Your Client With newBuilder()" to acknowledge caching images separately.

@chrisbanes your app, any preference?

Reply:
Hmmm, I think a different cache as Colin suggested. None of the requests should overlap (hopefully) so they should stay exclusive from each other.

Reply:
@yschimke Good point. I'll update the docs as well.